spec/requests/api/v3/statements_spec.rb

```diff
@@ -1,13 +1,36 @@
 require "rails_helper"
 
 RSpec.describe "Statements endpoint", type: "request" do
+  let(:current_lead_provider) { create(:lead_provider) }
+
   describe "GET /api/v3/statements" do
     context "when authorized" do
-      it "returns 200 - ok" do
-        api_get("/api/v3/statements")
+      context "when 2 statements exist for current_lead_provider" do
+        let!(:statement1) { create(:statement, lead_provider: current_lead_provider) }
+        let!(:statement2) { create(:statement, lead_provider: current_lead_provider) }
+
+        before do
+          create(:statement, lead_provider: create(:lead_provider, name: "Another lead provider"))
+        end
 
-        expect(response.status).to eq 200
-        expect(response.headers["Content-Type"]).to include("application/json")
+        it "returns 2 statements" do
+          api_get("/api/v3/statements")
+
+          expect(response.status).to eq 200
+          expect(response.content_type).to eql("application/json")
+          expect(parsed_response["data"].size).to eq(2)
+          expect(parsed_response["data"][0]["id"]).to eq(statement1.id)
+          expect(parsed_response["data"][1]["id"]).to eq(statement2.id)
```

Comment:
I'm not sure we should be testing the actual response here - shouldn't we trust our query object and serialiser? Otherwise we end up with request specs overlapping with both serialiser and query object ones.

The controller's responsibility here is just to respond to a request appropriately.

Reply:
I think its important to make sure the result is as we expect. In this case, we want to make sure the correct statements are returned, so only checked for the `id`'s. I shouldn't check all values as that's tested in the serializer unit tests.

Reply:
We end up in a situation where changes to the serializers or query objects mean we need to update API specs though

Reply:
We have to check whether we return the correct objects at least, otherwise its not really end to end.

Reply:
I was having a think about this, we need to exercise some form of reliable end to end testing, without going to deep into retesting the unit tests covered in serialisers/queries. I think checking size/ids is a good balance. If we stick to the following structure:
- Unit tests for serialisers
- Unit tests for queries and all edge cases
- Tests for ensuring schema is met for swagger
- Request specs which check statuses are ok, size and ids are ok and are very light happy path scenarios as in this PR it would be enough without delving into the fields returned and should cover us, otherwise I feel like we'll miss something if it does break